src/plugins/data/public/search/errors/utils.ts

```diff
@@ -15,6 +15,15 @@ export function getFailedShards(err: KibanaServerError<any>): FailedShard | unde
   return failedShards ? failedShards[0] : undefined;
 }
 
+function getNestedCause(err: KibanaServerError<any>): Reason {
```

Comment:
nit: you call `getNestedCause` from `getRootCause`. In parent method I see `KibanaServerError` instead of `KibanaServerError<any>`.

Reply:
Yes, thought about that.
But removing the `any` will set the generic `T` to `unknown` with types errors that need to be addressed. I think this is a major problem with EsErrors that should be addressed in another PR by providing better types.